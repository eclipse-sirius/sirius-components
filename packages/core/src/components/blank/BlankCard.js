/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/
import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { Card } from '../cards/Card';
import { Spacing } from '../spacing/Spacing';
import { M, L } from '../spacing/SpacingConstants';

import './BlankCard.css';

const propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

const BLANKCARD__CLASS_NAMES = 'blankcard';
const BLANKCARD_CONTAINER__CLASS_NAMES = 'blankcard-container';
const BLANKCARD_TITLE__CLASS_NAMES = 'title-l';

/**
 * The BlankCard component is used to render the empty state of a page.
 */
export const BlankCard = ({ className, title, message, ...props }) => {
  const blankCardClassNames = classNames(BLANKCARD__CLASS_NAMES, className);
  return (
    <Card {...props} className={blankCardClassNames}>
      <Spacing top={L} right={L} left={L} bottom={L}>
        <div className={BLANKCARD_CONTAINER__CLASS_NAMES}>
          <Spacing bottom={M}>
            <h1 className={BLANKCARD_TITLE__CLASS_NAMES}>{title}</h1>
          </Spacing>
          <p>{message}</p>
        </div>
      </Spacing>
    </Card>
  );
};
BlankCard.propTypes = propTypes;
