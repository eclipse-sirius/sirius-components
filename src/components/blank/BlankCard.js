/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/
import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { Body, Card } from '../cards/Card';

import './BlankCard.css';

const propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

const BLANKCARD__CLASS_NAMES = 'blankcard';
const BLANKCARD_TITLE__CLASS_NAMES = 'title-l';

/**
 * The BlankCard component is used to render the empty state of a page.
 */
export const BlankCard = ({ className, title, message, ...props }) => {
  const blankCardClassNames = classNames(BLANKCARD__CLASS_NAMES, className);
  return (
    <Card {...props} className={blankCardClassNames}>
      <Body>
        <h1 className={BLANKCARD_TITLE__CLASS_NAMES}>{title}</h1>
        <p>{message}</p>
      </Body>
    </Card>
  );
};
BlankCard.propTypes = propTypes;
