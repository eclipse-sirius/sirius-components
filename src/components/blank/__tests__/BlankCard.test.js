/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import Renderer from 'react-test-renderer';

import { BlankCard } from '../BlankCard';

describe('BlankCard', () => {
  it('renders a blank card', () => {
    const blankCard = Renderer.create(
      <BlankCard
        title="Nothing to see here"
        message="You should create new elements to see them here"
      />
    );
    expect(blankCard.toJSON()).toMatchSnapshot();
  });
});
