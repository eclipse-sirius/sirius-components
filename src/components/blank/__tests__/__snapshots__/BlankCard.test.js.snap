// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankCard renders a blank card 1`] = `
<div
  className="card blankcard"
>
  <div
    className="css-1nc71gt"
  >
    <div
      className="blankcard-container"
    >
      <div
        className="css-ie0fxa"
      >
        <h1
          className="title-l"
        >
          Nothing to see here
        </h1>
      </div>
      <p>
        You should create new elements to see them here
      </p>
    </div>
  </div>
</div>
`;
